### PR TITLE
fix: purge table statistic files

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix: purge table statistic files
add test case.

updated (by @dantengsky )

the main functionality of this PR is to purge table statistic files while doing `optimize table ... purge`.
pls correct me if I misunderstood @lichuang 

Closes https://github.com/datafuselabs/databend/issues/9054

### CODE REVIEW

File: src/query/storages/fuse/fuse/src/io/snapshots.rs [hunk 1]
```diff
@@ -102,6 +102,21 @@ impl SnapshotsIO {
             .map_err(|e| ErrorCode::StorageOther(format!("read snapshots failure, {}", e)))
     }
 
+    // Read all the snapshots by the root file.
+    // limit: read how many snapshot files
+    // with_segment_locations: if true will get the segments of the snapshot
```
Comment (dantengsky) [inline src/query/storages/fuse/fuse/src/io/snapshots.rs:107]: pls update the doc

File: src/query/storages/fuse/fuse/src/operations/gc.rs [hunk 2]
```diff
@@ -131,6 +138,29 @@ impl FuseTable {
...
+        {
+            if let Some(root_ts_location) = root_ts_location_opt {
+                let start = Instant::now();
+                let snapshots_io = SnapshotsIO::create(
+                    ctx.clone(),
+                    self.operator.clone(),
+                    self.snapshot_format_version().await?,
+                );
+                let ts_to_be_purged = snapshots_io
+                    .read_table_statistic_files(&root_ts_location, None)
```
Comment (dantengsky) [inline src/query/storages/fuse/fuse/src/operations/gc.rs:151]: will the `root_ts` be included in the `ts_to_be_purged`?

**Comment by dantengsky** [comment]
@lichuang  summary of PR has been updated, please correct it if wrongly described.

File: src/query/storages/fuse/fuse/src/operations/gc.rs (hunk 2 above)
Comment (lichuang) [reply src/query/storages/fuse/fuse/src/operations/gc.rs:151]: no, exclude the `root_ts` file.

**Review by dantengsky — APPROVED** [review]

**Pull request merged**